src/components/MoviePages/MainPage.vue

```diff
@@ -42,22 +63,148 @@ const options = {
   }
 };
 
+const fetchTrendMovies = async () => {
+  try {
+    await new Promise(resolve => setTimeout(resolve, 2000));
+    const response = await fetch('https://api.themoviedb.org/3/movie/popular?language=en-US&page=1', options);
+    if (!response.ok) {
+      throw new Error('This URL does not exist');
+    }
+    const data = await response.json();
+    popularMovies.value = data.results;
+  } catch (err) {
+    console.error(err);
+  }
+};
+
+
+ const fetchMovies = async () => {
+  try {
+    await new Promise(resolve => setTimeout(resolve, 2000));
+    const response = await fetch('https://api.themoviedb.org/3/trending/movie/day?language=en-US', options);
+    if (!response.ok) {
+      throw new Error('This URL does not exist');
+    }
+    const data = await response.json();
+    // mainMovies.value = data.results;
+
+    const genereResponse = await fetch('https://api.themoviedb.org/3/genre/movie/list?language=en', options)
+    const genereData = await
+     genereResponse.json()
+    
+    const genreMap = {};
+    genereData.genres.forEach(genre => {
+      genreMap[genre.id] = genre.name;
+    });
+    
+
+    // Assign genre names to each movie
+    const moviesWithGenres = data.results.map(movie => {
+      const genreNames = movie.genre_ids.map(genreId => genreMap[genreId]);
+      return {
+        ...movie,
+        genre_names: genreNames,
```

Comment:
no need for this function...
you can do. this with a simple Map when you showing the component